File: SpaceLegend/Scripts_csharp/ScriptPlayer.cs
using UnityEngine;
using System.Collections;

public class ScriptPlayer : MonoBehaviour {
	
public float speed = 10.0f;
public Transform disparo; //esto carga el disparo
public Transform arma1; //esto carga el arma1
 
	
	// Use this for initialization
	void Start () {
	
	}
	
	// Update is called once per frame
	void Update () 
	{
	//movimiento básico
	float transV = Input.GetAxis ("Vertical") * speed * Time.deltaTime;
	float transH= Input.GetAxis ("Horizontal") * speed * Time.deltaTime;
		
	transform.Translate(transH,transV,0);
	//limitaciones
		
   	if(transform.position.x >= 12)
		{
   		 transform.position = new Vector3 (12, transform.position.y, transform.position.z);
		}
  	if(transform.position.x <= -12)
		{
   		transform.position = new Vector3 (-12, transform.position.y, transform.position.z);
		}
   	if(transform.position.y >= 9)
		{
   		transform.position = new Vector3 (transform.position.x, 9, transform.position.z);
		}
   	if(transform.position.y <= -9)
		{
   		transform.position = new Vector3 (transform.position.x, -9, transform.position.z);
		}
	
	//definimos el disparo
   	if(Input.GetKeyDown("space"))
	   	{
	   	Instantiate(disparo, arma1.position, arma1.rotation);
	   	}
	}
}

File: SpaceLegend/Scripts_js/scriptPlayer.js
#pragma strict
var speed   : float = 10.0;
var shoot : Transform; 
var weapon1   : Transform; 
var weapon2   : Transform; 
var weapon3   : Transform; 
var vida : int;
var RightJoystick : Joystick;
var LeftJoystick : Joystick;
 
function Start () {

}

function Update () {
    
    //basic movement
    var transV : float = Input.GetAxis ("Vertical") * speed * Time.deltaTime;
    var transH : float = Input.GetAxis ("Horizontal") * speed * Time.deltaTime;
    
   	var currentPosition = transform.TransformPoint(Vector3.zero); 
   	
   	//limits
	if(transform.position.x >= 12)
   		transform.position = currentPosition;
   	if(transform.position.x <= -12)
   		transform.position = currentPosition;
   	if(transform.position.y >= 12)
   		transform.position.y = 12;
   	if(transform.position.y <= -12)
		   transform.position.y = -12;   		
		   
   	//set shoot
   	if ( RightJoystick.tapCount == 1 )
   	{
   	if (Application.loadedLevelName == "1"||Application.loadedLevelName == "2" )
   	 {
   	 	Instantiate(shoot, currentPosition, shoot.rotation);
   	  }
   	 if (Application.loadedLevelName == "3"||Application.loadedLevelName == "4" )
   	 {
   		Instantiate(shoot, weapon1.position, shoot.rotation);
   		Instantiate(shoot, weapon2.position, shoot.rotation);
	 }
	 if (Application.loadedLevelName == "5"||Application.loadedLevelName == "6" )
   	 {
   		Instantiate(shoot, weapon1.position, shoot.rotation);
   		Instantiate(shoot, weapon2.position, shoot.rotation);
   		Instantiate(shoot, weapon3.position, shoot.rotation);
	 }
   	audio.Play();
   	}
}


